xdsl/dialects/comb.py

```diff
@@ -0,0 +1,245 @@
+from typing import Annotated
+
+from xdsl.dialects.builtin import IndexType, IntegerAttr, IntegerType, UnitAttr, i32
+from xdsl.ir import Dialect, OpResult
+from xdsl.irdl import (
+    ConstraintVar,
+    IRDLOperation,
+    Operand,
+    VarOperand,
+    attr_def,
+    irdl_op_definition,
+    operand_def,
+    opt_attr_def,
+    result_def,
+    var_operand_def,
+)
+
+
+class BinCombOp(IRDLOperation):
+    """
+    A binary comb operation. It has two operands and one
+    result, all of the same integer type.
+    """
+
+    T = Annotated[IntegerType, ConstraintVar("T")]
+
+    lhs: Operand = operand_def(T)
+    rhs: Operand = operand_def(T)
+    result: OpResult = result_def(T)
+
+    two_state: UnitAttr | None = opt_attr_def(UnitAttr)
+
+
+class VariadicCombOp(IRDLOperation):
+    """
+    A variadic comb operation. It has a variadic number of operands, and a single
+    result, all of the same type.
+    """
+
+    T = Annotated[IntegerType, ConstraintVar("T")]
+
+    inputs: VarOperand = var_operand_def(T)
+    result: OpResult = result_def(T)
+
+    two_state: UnitAttr | None = opt_attr_def(UnitAttr)
```

Comment:
Can you please add documentation for this variable?

Reply:
Which one? The two_state variable?

Reply:
yes, please, it's not obvious to me just by the name what it's supposed to do